astroid/nodes/node_classes.py

```diff
@@ -242,693 +237,6 @@ def _container_getitem(instance, elts, index, context=None):
     raise AstroidTypeError("Could not use %s as subscript index" % index)
 
 
-OP_PRECEDENCE = {
-    op: precedence
-    for precedence, ops in enumerate(
-        [
-            ["Lambda"],  # lambda x: x + 1
-            ["IfExp"],  # 1 if True else 2
-            ["or"],
-            ["and"],
-            ["not"],
-            ["Compare"],  # in, not in, is, is not, <, <=, >, >=, !=, ==
-            ["|"],
-            ["^"],
-            ["&"],
-            ["<<", ">>"],
-            ["+", "-"],
-            ["*", "@", "/", "//", "%"],
-            ["UnaryOp"],  # +, -, ~
-            ["**"],
-            ["Await"],
-        ]
-    )
-    for op in ops
-}
-
-
-class NodeNG:
```

Comment:
Maybe we should add it in `astroid.nodes` as it can be used for typing

Reply:
+1 It would even make sense to add it to `nodes.py` for `2.6.0` with a deprecation note in the `Changelog` for accessing it through `node_classes.py`

Reply:
Let's do this in two steps :
Step 1 : https://github.com/PyCQA/astroid/pull/1067
Step 2 : This MR

Reply:
I think it can go in 2.6.0 as we do not have breaking changes anymore.

Reply:
> Let's do this in two steps ...

+1

> I think it can go in 2.6.0 as we do not have breaking changes anymore.

Just doing #1067 doesn't mean that this MR isn't a breaking change anymore. Even if #1067 is released with `2.6`. Additionally, I would really love it if we move these big refactorings to `2.7`, just to make sure we / I didn't break stuff previously. It should be fine, but you never know unfortunately. It will be a lot easier to do a bug fix then.

Reply:
Sure, there's https://github.com/PyCQA/astroid/pull/935 to review before releasing 2.6, could you check it ? I think we could release 2.6 as soon as 935 is merged. This way we can check everything in pylint and release 2.9.0 😄 

Regarding this MR and #1067 I added deprecation warning and the old API in this MR (https://github.com/PyCQA/astroid/pull/1057/commits/09a8ea5501d67e8515396c4ea4c0be6793d1028d) so I think it's safe and not a breaking change but we can play it safe and push it to 2.7 to not delay pylint. I would hate for those refactor to be stuck for a long time as each change in ``NodeNg`` will create a conflict and I'd need to redo it from sratch.

Reply:
> Sure, there's #935 to review before releasing 2.6, could you check it ?

Just did. I don't think that needs to go into `2.6` though as it isn't a fix. At least that's how I understand it. I would just consider #1057 and #1067 for the release. Running some final tests on #1067 at the moment.

> Regarding this MR and #1067 I added deprecation warning and the old API in this MR (09a8ea5) so I think it's safe and not a breaking change but we can play it safe and push it to 2.7 to not delay pylint. I would hate for those refactor to be stuck for a long time as each change in NodeNg will create a conflict and I'd need to redo it from sratch.

I understand the issue. However, doing it now will cause even bigger problems if really we need to do a hotfix. 🤞🏻 we don't. I'm not planning to open any new MR for astroid for the moment.

Reply:
Well I wanted to merge 935 because it looked easy to review and ready but you're right, we don't have to. 😄 

I'm going to finalize #1057 then.

Reply:
I think I mixed up the MR numbers 🤦🏻 It should have been #1062 and #1067, not this one. So only #1062 would be left for the release.